packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/_get_type.py

```diff
@@ -29,8 +31,13 @@
 }
 
 
-def get_type_default(type: Field.Kind.ValueType, repeated: bool) -> Any:
+def get_type_default(type: Field.Kind.ValueType, repeated: bool, default_value: Any = None) -> Any:
     """Get the default value for the give type."""
     if repeated:
         return list()
+    if isinstance(default_value, list):
+        default_value = default_value[0]
```

Comment:
Oh, I see. You're passing in the default value of the repeated field when querying its element type. You could pass in default_value[0] here.

```
    if parameter_metadata.repeated:
        expected_element_type = type(
            get_type_default(parameter_metadata.type, False, default_value)
        )
```

Reply:
It's for the message arrays where we get the expected_element_type, removing those lines make the function return a list instead of the class of the default value. 

If we pass in the first element of the default value it would make make this test fail where the default value is a empty string, giving a index error

`expected_element_type = type(get_type_default(parameter_metadata.type, False, default_value[0))`
`test___measurement_service___add_configuration_with_mismatch_default_value__raises_type_error`

Reply:
That's one of those quirky things about Python: there is no `char` data type, and indexing a `str` gives you a one-character `str`.

There's another corner case you need to handle here: empty lists.

How about changing `_validate_default_value_type` to do something like this?
```
if parameter_metadata.repeated:
    if not isinstance(default_value, list):
        raise TypeError(...)
    if len(default_value) > 0:
        expected_element_type = type(
            get_type_default(parameter_metadata.type, False, default_value[0])
        )
        ...
```